flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/Executing.java

```diff
@@ -47,6 +51,10 @@
 class Executing extends StateWithExecutionGraph implements ResourceListener {
 
     private final Context context;
+    private Instant lastRescale = Instant.EPOCH;
```

Comment:
Actually, I don't understand the `lastRescale` of your implematation. The `Executing` is a new object after each rescale, so `lastRescale` isn't used across multiple rescale.

We should record the `Instant.now()` as the lastRescale at the constructor, right?

Reply:
I agree, there is indeed a bug in Executing being a new object with each rescale when the transition to Restaring and then to Executing state is done. I thought the state of the Executing object was kept between the restarts (with checkpointing) but it is not, a new one is created in the transitionToState(). This leeds to lastRescale being always be EPOCH. Thanks for pointing out.

Reply:
recording lastRescale as Instant.now() in the Executing constructor could work but it means that even a regular transition from Created to Executing state (see [FLIP-160](https://cwiki.apache.org/confluence/display/FLINK/FLIP-160%3A+Adaptive+Scheduler)) would be considered a rescale.

Reply:
Fixed